Release/src/http/client/http_client_asio.cpp

```diff
@@ -252,7 +255,10 @@ class asio_connection_pool
     m_io_service(io_service),
     m_timeout_secs(static_cast<int>(idle_timeout.count())),
     m_start_with_ssl(start_with_ssl),
-    m_ssl_context_callback(ssl_context_callback)
+    m_ssl_context_callback(ssl_context_callback),
+    m_pool_timeout_secs(60), // Clean this connection pool 60 secs after the last asio_client release it.
+    m_pool_timer(io_service),
+    m_use_count(0)
```

Comment:
Manual reference counting is extremely error prone; I understand that to accomplish the exact cleanup design you've specified (60 seconds after last asio_client) it seems needed, but this alone is a reason to change the cleanup design.

Instead of having a separate timer, what about having the pool automatically cleaned up when the last connection times out? That should be as simple as "whenever removing a connection from a connection_pool, check if the set of connections for this endpoint is now empty. If so, remove the entry for this endpoint".